src/track/keyutils.cpp

```diff
@@ -325,19 +380,32 @@ ChromaticKey KeyUtils::guessKeyFromText(const QString& text) {
 
         QString scale = keyMatcher.cap(3);
         // we override major if a scale definition exists
-        if (! scale.isEmpty()) {
+        if (!scale.isEmpty()) {
             if (scale.compare("m", Qt::CaseInsensitive) == 0) {
+                // Aeolian
                 major = false;
             } else if (scale.startsWith("min", Qt::CaseInsensitive)) {
+                // Aeolian
                 major = false;
             } else if (scale.startsWith("maj", Qt::CaseInsensitive)) {
+                // Ionian
                 major = true;
             } else {
-                qDebug() << "WARNING: scale from regexp has unexpected value."
-                  " should never happen";
+                // Try to find detailed scale mode
+                ScaleMode scaleMode = ScaleMode::Unknown;
+                for (size_t i = 0; i < std::size(s_scaleModeTextShort); ++i) {
+                    if (scale.startsWith(s_scaleModeTextShort[i], Qt::CaseInsensitive)) {
+                        scaleMode = static_cast<ScaleMode>(i);
+                        break;
+                    }
+                }
+                if (scaleMode == ScaleMode::Unknown) {
+                    return mixxx::track::io::key::INVALID;
+                }
+                major = s_scaleModeIsMajor[static_cast<int>(scaleMode)];
+                steps += s_scaleModeTransposeSteps[static_cast<int>(scaleMode)];
```

Comment:
I personally don't like all the hidden dependencies on the sizes and positions this has...
IMO a first step to making this better would be to create a utility method for this that lives close to all these arrays. Then its less likely that some reorder as part of a refactoring would break this code 300 lines later.

Reply:
I have created a position independent struct. A additional function that wraps this is no longer required.

Reply:
Thank you. much better now.